packages/data/src/world/mq/deku_tree_mq.yml

```diff
@@ -2,126 +2,282 @@
   dungeon: DT
   exits:
     "Kokiri Forest Near Deku Tree": "true"
-    "Deku Tree Lobby": "true"
-"Deku Tree Lobby":
+    "MQ Deku Tree Lobby": "true"
+"MQ Deku Tree Lobby":
   dungeon: DT
   exits:
-    "Deku Tree Compass Room": "can_use_bow || (can_use_slingshot && (can_use_sticks || can_use_din))"
-    "Deku Tree Water Room": "(can_use_slingshot || can_use_bow) && (can_use_sticks || has_fire)"
-    "Deku Tree Basement Ledge": "is_adult || event(DEKU_BLOCK) || trick(OOT_DEKU_SKIP) || has_hover_boots"
+    "Deku Tree": "true"
+    "MQ Deku Tree 2nd Floor": "true"
+    "MQ Deku Tree Basement": "has_fire || (can_use_sticks && event(DEKU_MQ_MAIN_TORCH))"
   events:
-    STICKS: "has_weapon || can_boomerang"
+    STICKS: "can_kill_baba_sticks"
     NUTS: "can_kill_baba_nuts"
   locations:
-    "MQ Deku Tree Map Chest": "true"
-    "MQ Deku Tree Slingshot Chest": "soul_deku_baba && soul_enemy(SOUL_ENEMY_GOHMA_LARVA) && (has_weapon || can_use_sticks || has_ranged_weapon_child)" #Bombs and Din's work, but it's horrible.
-    "MQ Deku Tree Slingshot Room Far Chest": "can_use_sticks || has_fire"
-    "MQ Deku Tree Basement Chest": "can_use_sticks || has_fire"
-    "MQ Deku Tree GS Lobby Crate": "gs && can_damage_skull"
     "MQ Deku Tree Grass Entrance Lower 1": "can_cut_grass"
     "MQ Deku Tree Grass Entrance Lower 2": "can_cut_grass"
     "MQ Deku Tree Grass Entrance Lower 3": "can_cut_grass"
     "MQ Deku Tree Grass Entrance Lower 4": "can_cut_grass"
     "MQ Deku Tree Grass Entrance Lower 5": "can_cut_grass"
+"MQ Deku Tree 2nd Floor":
+  dungeon: DT
+  exits:
+    "MQ Deku Tree Lobby": "true"
+    "MQ Deku Tree 3rd Floor": "true"
+    "MQ Deku Tree Room Before Compass Room Entry Door": "has_fire || (can_use_sticks && event(DEKU_MQ_MAIN_TORCH)) || (event(DEKU_MQ_MAIN_TORCH) && can_use_bow)"
+  locations:
+    "MQ Deku Tree Map Chest": "true"
+    "MQ Deku Tree GS Lobby Crate": "gs && can_damage_skull"
```

Comment:
for now this is fine, but hook, bow, and slingshot would need to be able to break the crate.

Reply:
Not true. In MQ, spiders are under crates, not spawned from breaking them. If you kill this spider, item appears on top of crate.

Reply:
Still true. how are you gonna kill the spider if you can't shoot through the crate

Reply:
I added a note for hookshot, as I can't seem to replicate it. But I did do it before when I wrote the logic.